htdocs/core/lib/functions.lib.php

```diff
@@ -1689,7 +1689,11 @@ function dol_substr($string,$start,$length,$stringencoding='')
 	if (empty($stringencoding)) $stringencoding=$langs->charset_output;
 
 	$ret='';
-	if (function_exists('mb_substr'))
+	if (function_exists('iconv_substr'))
+	{
+		$ret=iconv_substr($string,$start,$length,$stringencoding);
+	}
+	else if (function_exists('mb_substr'))
```

Comment:
It was just a comment, of course it is not important at all because there is actually a mix of each one's coding style all over Dolibarr.